### PR TITLE
Add docs about ecosystem

### PR DESCRIPTION
**Pull request by leplatrem**
@n1k0 any feedback would be welcome :) (*API wording, etc.*)

### CODE REVIEW

File: docs/ecosystem.rst [hunk 1]
```diff
@@ -0,0 +1,206 @@
+Ecosystem
+#########
+
+.. note::
+
+    If you build a package that you would like to see listed here, just
+    get in touch with us !
```
Comment (n1k0) [inline docs/ecosystem.rst]: Nit: no space before `!` in English

File: docs/ecosystem.rst [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+Appart from usual python «*import and use*», *Pyramid* can include external
+packages, that can bring views, event listeners etc.
+
+.. code-block:: python
+    :emphasize-lines: 7
+
+    def main(global_config, **settings):
+        config = Configurator(settings=settings)
+
+        cliquet.initialize(config, __version__)
```
Comment (n1k0) [inline docs/ecosystem.rst]: Nit: would be useful to have the `cliquet` import part of that snippet

File: docs/ecosystem.rst [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    # myproject.ini
+    cliquet.cache_backend = cliquet_riak.cache
+
+
+Adding features
+===============
+
+Another use-case would be to add extra-features, like indexing for example.
+
+* Initialize an indexer ;
```
Comment (n1k0) [inline docs/ecosystem.rst]: Nit: no space before `;` in English

File: docs/ecosystem.rst [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+            indexer = self.request.registry.indexer
+            indexer.index_record(self.name, record)
+
+            return r
+
+.. note::
+
+    In this example, ``IndexedResource`` is inherited, and must hence be
+    used explicitly as base resource class in applications.
```
Comment (n1k0) [inline docs/ecosystem.rst]: Nit: s/as base/as a base/

File: docs/ecosystem.rst [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    A nicer pattern would be to trigger *Pyramid* events in *Cliquet* and
+    let packages like this one plug listeners. If you're interested,
+    `we started to discuss it <https://github.com/mozilla-services/cliquet/issues/32>`_ !
```
Comment (n1k0) [inline docs/ecosystem.rst]: Nit: no space before `!`

File: docs/ecosystem.rst [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        // success !
+      });
+
+    articles.get('id-1234')
+      .then(function (record) {
+        // Read from local if offline.
+      });
+
+    articles.filter({
+        'title': {'$eq': 'Hello'}
```
Comment (n1k0) [inline docs/ecosystem.rst]: Nit: no need to enclose attribute names here

File: docs/ecosystem.rst [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+      })
+      .then(function (results) {
+        // List of records.
+      });
+
+    articles.sync()
+      .then(function (result) {
+        // Synchronize offline store with server.
+      });
```
Comment (n1k0) [inline docs/ecosystem.rst]: We should probably highlight how we deal with errors, eg using `.catch`

**Comment by n1k0** [comment]
Looks good. Some minor comments you might want to address before landing.

File: docs/ecosystem.rst [hunk 8]
```diff
@@ -0,0 +1,254 @@
+Ecosystem
+#########
+
+.. note::
+
+    If you build a package that you would like to see listed here, just
+    get in touch with us!
+
+
```
Comment (ametaireau) [inline docs/ecosystem.rst]: I would like to see here an introduction of what this document is providing.

File: docs/ecosystem.rst [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+Basics about Pyramid apps
+=========================
+
+Include external packages
```
Comment (ametaireau) [inline docs/ecosystem.rst]: I wonder if we should put this under "extending cliquet".

**Pull request merged**